on-demand_downscaling/helpers.py

```diff
@@ -361,9 +391,137 @@ def handle_run_downscaling(arg):
     else:
         approx_time = 12 if period.value == "1950-2010" else 24
     print(f"Approximate time to completion: {str(approx_time)} minutes")
-    global downscaled_outputs
-    downscaled_outputs[gcm_var].append(chickadee.ci(**chickadee_params))
-    print()
+
+    try:
+        with capture_output() as captured:
+            ci_process = chickadee.ci(**chickadee_params)
+
+        # Retrieve the UUID
+        status_url = ci_process.statusLocation
+        process_uuid = urlparse(status_url).path.split("/")[-1].replace(".xml", "")
+        print(f"Downscaling process UUID: {process_uuid}")
+        birdy_widget = captured.outputs[0].data[
+            "application/vnd.jupyter.widget-view+json"
+        ]["model_id"]
+        widget_instance = Widget.widgets[birdy_widget]
+        cancel_button = find_cancel_button(widget_instance)
+
+        if cancel_button:
+            # Hide cancel button initially
+            cancel_button.layout.display = "none"
+
+            def check_process_status():
+                try:
+                    status_response = requests.get(status_url)
+                    if status_response.status_code == 200:
+                        status_text = status_response.text
+                        if "ProcessStarted" in status_text:
+                            cancel_button.layout.display = "block"
+                            return False  # Stop
+                        elif (
+                            "ProcessSucceeded" in status_text
+                            or "ProcessFailed" in status_text
+                        ):
+                            return False  # Stop
+                    return True  # Continue
```

Comment:
Does a non 200 error state trigger the exception below to exit? For example if, for some reason, the URL 404ed, would this just spin forever?

I'd probably recommend guarding against dangling spinning threads with something like either a max polls, max failures or max duration in the loop.